db/low-level/src/ts/dns.ts

```diff
@@ -0,0 +1,241 @@
+    // How that is defined, I have no clue, but try to use real words no random capitals
+    public async register(longName: string): Promise<void> {
+        const response = await saneResponse(WebRequest.post(
+            this.mkendpoint(`/${longName}`), {
+                auth: {
+                    bearer: (await this.authRes).token
+                }
+            }));
+
+        if (response.statusCode !== 200) {
+            throw new SafeError(`statusCode=${response.statusCode} !== 200`, response);
+        }
+    }
+
+    /** @arg longName - public name that can be shared
+     *  @arg serviceName - name of service mapped to longName
+     *  @arg rootPath - app or drive
+     *  @arg serviceHomeDirPath - the full path of the directory to be associated with to the service
+     *  @returns a promise to say if the longName and service were registered
+     */
+    public async registerAndAddService(longName: string, serviceName: string,
+                                       rootPath: RootPath, serviceHomeDirPath: string): Promise<void> {
+        let Body = {
+            "longName": longName,
+            "serviceName": serviceName,
+            "rootPath": rootPath,
+            "serviceHomeDirPath": serviceHomeDirPath
+        }
+
+        const response = await saneResponse(WebRequest.post(
+            this.mkendpoint(""), {
+                auth: {
+                    bearer: (await this.authRes).token
+                },
+                json: true,
+                body: Body
+            }));
+
+        if (response.statusCode !== 200) {
+            throw new SafeError(`statusCode=${response.statusCode} !== 200`, response);
+        }
+    }
+
+    /** @arg longName - public name that can be shared
+     *  @returns a list of services mapped to the longName
+     */
+    public async getServices(longName: string): Promise<DnsServiceList> {
+        const response = await saneResponse(WebRequest.get(
+            this.mkendpoint(`/${longName}`), {
+                auth: {
+                    bearer: (await this.authRes).token
+                }
+            }));
+
+        if (response.statusCode !== 200) {
+            throw new SafeError(`statusCode=${response.statusCode} !== 200`, response);
+        }
+
+        return JSON.parse(response.content); //little sketchy
+    }
+
+    /** @arg longName - public name that can be shared
+     *  @arg serviceName - name of service mapped to longName
+     *  @arg rootPath - app or drive
+     *  @arg serviceHomeDirPath - the full path of the directory to be associated with to the service
+     *  @returns a promise to say if the longName and service were registered
+     *
+     *  TODO: (Abdi) maybe make this a generic function for the one above?
+     */
+
+    public async addService(longName: string, serviceName: string,
+                            rootPath: RootPath, serviceHomeDirPath: string): Promise<void> {
+
+        let Body = {
```

Comment:
annoying nit: I think Body could be `const` here. Don't feel like you have to fix this to check in, but if you are making some other changes, why not?